16.12.23

Aleksei-Nizhegorodov opened the pull request:
- Добавил ООП - класс

Aleksei-Nizhegorodov committed:
1. Вывести расчётный код в отдельную функцию – солвер:
2. Логика решения методом характеристик
созадть функцию, которая реализовывает метод характеристик
3. Вывод расчета отдельной ф-цией
4. documentation comments style = Doxygen (///)

Aleksei-Nizhegorodov committed:
Исправлены ошибки в процессе PR

Aleksei-Nizhegorodov committed:
добавлен класс

Aleksei-Nizhegorodov committed:
добавлен класс

alexbibich commented on ConsoleApplication16.cpp:
```diff
@@ -1,116 +1,124 @@
+            dt = dx / iniPipe.speed;
+            num_layers = (T / dt);
+        }
+    };
```
Создание структур должно быть вне класса, далее уже внутрь класса можно передавать и хранить экземпляры этих структур

alexbibich commented on ConsoleApplication16.cpp:
```diff
@@ -1,116 +1,124 @@
+                calculation(iniPipe, sera_in[i], buffer.current()[j], buffer.previous()[j]);
+            }
+            buffer.advance(1);
+            output_fun(iniCalc, buffer.current(), i);
```
Строчки с 107 до 113 надо оставить в этом методе класса, назвать его говорящим именем, остальное вернуть в main. Профили передавать в во внутренние переменные класса через конструктор

alexbibich commented on ConsoleApplication16.cpp:
```diff
@@ -1,116 +1,124 @@
+                }
+            }
+            outFile.close();
+        }
```
Класс, выполняющий все функции, называется Божественным объектом (God class - посмотри в гугле). От такого надо уходить, один класс направлен на выполнение определённых функций, в данном случае - решение методом характеристик. Отсюда следует, что функцию вывода в файл надо вынести вне класса

Aleksei-Nizhegorodov committed:
- Вынесены структуры за класс
- Профили передаю в во внутренние переменные класса через конструктор
- Функцию вывода в файл вынес вне класса

alexbibich commented on ConsoleApplication16.cpp:
```diff
@@ -1,116 +1,144 @@
+/// @brief Структура профилей вхлдящих параметров плотности
+struct RoProfile {
+    vector<double> data;
+};
```
Зачем заводить структуры под один вектор? Не имеет смысла, используй просто вектор

alexbibich commented on ConsoleApplication16.cpp:
```diff
@@ -1,116 +1,144 @@
+/// @brief Структура профилей входящих параметров серы
+struct SeraProfile {
+    vector<double> data;
+};
```
Аналогично, убери структуры и в коде используй просто вектор

alexbibich commented on ConsoleApplication16.cpp:
```diff
@@ -1,116 +1,144 @@
-        Overview << iniCalc.dt * (j + 1) << "," << 0 << "," << ro_1[0] << "," << sera_1[0] << endl; // Запись  в файл
+    // Профили входящих значений
+    RoProfile roProfile = { {  850,  870,  860,  860,  860,  860,  860,  860,  860,  870 } };
+    SeraProfile seraProfile = { {  0.01,  0.02,  0.03,  0.04,  0.05,  0.06,  0.07,  0.08,  0.09,  0.01 } };
```
Сделать просто векторами

Aleksei-Nizhegorodov committed:
Исправлены замечания с профилями входящих значений

victorsouth commented:
С замечаниями в целом согласен. 
Если расчет в текущем виде работает, предлагаю этот PR закрывать, продолжить в новой ветке.

victorsouth commented:
По рекомендации Александра предлагаю завершать Блока 1 у Алексея
В данном PR устранить конфликты, смержить в мастер.

Aleksei-Nizhegorodov committed:
Merge branch 'master' into 16.12.23
